artemis/modules/dangling_dns_detector.py

```diff
@@ -0,0 +1,201 @@
+import ipaddress
+import socket
+from typing import Any
+
+import dns.message
+import dns.name
+import dns.query
+import dns.resolver
+from dns import rdatatype
+from dns.rdata import Rdata
+from karton.core import Task
+
+from artemis import load_risk_class
+from artemis.binds import TaskStatus, TaskType
+from artemis.module_base import ArtemisBase
+from artemis.task_utils import get_target_host
+
+
+def direct_dns_query(
+    name: dns.name.Name, record_type: rdatatype.RdataType, server_ip: ipaddress.IPv4Address
+) -> dns.message.Message | None:
+    try:
+        query = dns.message.make_query(name, record_type)
+        response = dns.query.udp(query, str(server_ip), timeout=5)
+
+        return response
+    except Exception:
+        return None
+
+
+def edns_query(target: str, record_type: rdatatype.RdataType) -> dns.resolver.Answer | None:
+    try:
+        resolver = dns.resolver.Resolver()
+        resolver.use_edns(True)
+        return resolver.resolve(target, record_type)
+    except Exception:
+        return None
+
+
+def ip_exists(ip: str, port: int = 80, timeout: float = 2.0) -> bool:
```

Comment:
I'd add a larger timeout and check more ports (maybe also ICMP). FPs hurt us more than longer module runtime